food-waste-ui/src/components/Basket/Basket.jsx

```diff
@@ -0,0 +1,39 @@
+import * as React from "react";
+import { Link } from "react-router-dom";
+import axios from "axios";
+import "./Basket.css";
+
+export default function Basket({ currentUser }) {
+  const [basket, setBasket] = React.useState([]);
+  const [isLoading, setIsLoading] = React.useState(false);
+  const currentUserId = localStorage.getItem("current_user_id");
+
+  React.useEffect(() => {
+    setIsLoading(true);
+    axios.get("http://localhost:3001/makeapost").then((response) => {
+      let allProducts = response.data.products;
+      let allProductsInBasket = [];
+      for (let i = 0; i < allProducts.length; i++) {
```

Comment:
Is there a way we can do this without a nested for loop? Maybe storing another data struct? 

Is allProducts all the products the market place has within a specific radius? How would this work if we scale larger and many many products are available for users to choose from?